Libraries/src/Amazon.Lambda.Serialization.SystemTextJson/Converters/LongToStringJsonConverter.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Amazon.Lambda.Serialization.SystemTextJson.Converters
+{
+    public class LongToStringJsonConverter : JsonConverter<string>
```

Comment:
This converter makes me nervous. It will run for every `string` value that is serialized. Since this is just for the `State` property for Kinesis we shouldn't add this to the global list of converters we run. 

I suggest we:
* Change the converter to use `Dictionary<string, string>` instead of `string` as the generic and then implement for a Dictionary.
* Rename it to `DictionaryLongToStringJsonConverter`
* Instead of adding the converter to the global list decorate the `State` attributes in the Kinesis POCOs with the `JsonConverter` attribute. That way this converter is only used on the scenarios where we know the special scenario exists.
* Update the documentation on the Kinesis POCOs informing users that they are only compatible with the serializers using System.Text.json unless you intend to also write the equivalent for Newtonsoft.

Reply:
Done for most part. I don't think we need to update the documentation (for last point) since it should work.